src/core/compiler/CircomCompilerDownloader.ts

```diff
@@ -84,18 +84,18 @@ export class CircomCompilerDownloader {
     const compilerBinaryPath = this._getCompilerDownloadPath(version);
     const wasmCompilerBinaryPath = this._getWasmCompilerDownloadPath(version);
 
-    if (await fs.pathExists(compilerBinaryPath)) {
-      return { binaryPath: compilerBinaryPath, version: version, isWasm: false };
-    }
-
     if (await fs.pathExists(wasmCompilerBinaryPath)) {
       return { binaryPath: wasmCompilerBinaryPath, version: version, isWasm: true };
     }
 
+    if (await fs.pathExists(compilerBinaryPath)) {
+      return { binaryPath: compilerBinaryPath, version: version, isWasm: false };
+    }
```

Comment:
Is it intensional to return `wasm` first?

Reply:
Yes, this is needed when compilerBinaryPath and wasmCompilerBinaryPath are the same (both pointing to circom.wasm), so that isWasm flag is set properly.
It can happen when we fallback to WASM as a platform in case of an unsupported os platform